File: TS/Basic/Collections/sets.ts
import assert from 'assert/strict';

export function setsExample() {
  //Create a Set
  let diceEntries = new Set<number>([1, 2, 3, 4, 5]);
  //Add values
  diceEntries.add(6);
  //Check value is present or not
  assert(diceEntries.has(1));
  assert(!diceEntries.has(10)); //6
  //Size of Set
  console.log("size = ", diceEntries.size);
  //Delete a value from set
  console.log(diceEntries);
  diceEntries.delete(6); // true
  console.log("after delete 6 ", diceEntries);
  //Clear whole Set
  diceEntries.clear();
  console.log("after clear ", diceEntries);
}

File: Nim/old/Collections/sets.nim
import std/sets


# Create a Set
var diceEntries = toHashSet([1, 2, 3, 4, 5])
# Add values
diceEntries.incl 6
# Check value is present or not
assert diceEntries.contains 5
assert not diceEntries.contains 10
# Size of Set
echo "size = ", diceEntries.len
# Delete a value from set
echo diceEntries
diceEntries.excl 6
echo "after delete 6 ", diceEntries
# Clear whole Set
diceEntries.clear()
echo "after clear ", diceEntries